.github/workflows/gha-ci.yml

```diff
@@ -0,0 +1,103 @@
+name: GitHubActions CI tests
+
+on:
+  push:
+    branches: [master, dev, ci]
+  pull_request:
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, macos-10.15]
+        build_suite: [autoconf]
+        with_coveralls: [no]
+        include:
+          # Do one build with cmake instead of autoconf
+          - os: ubuntu-20.04
+            build_suite: cmake
+            with_coveralls: no
+          # And run one build with gcov and coveralls
+          # (Does currently not work, but neither did it in travis)
+          #- os: ubuntu-20.04
+          #  build_suite: autoconf
+          #  with_coveralls: yes
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+
+      - name: install deps
+        run: |
+          if (echo "${{ matrix.os }}" | grep -qE '^macos-') ; then
+            brew update
+            # freetype, gettext, icuc4 and autoconf are preinstalled
+            # and `brew install` fails if a non-uptodate version is already installed
+            #brew upgrade autoconf freetype gettext icu4c
+            brew install automake boost ffmpeg ffms2 fftw fribidi libass \
+                         lua luarocks m4 wxmac hunspell uchardet
+          else
+            #sudo apt-get update && sudo apt-get upgrade
+            sudo apt-get install -y ${{ matrix.build_suite }} \
+              autopoint build-essential g++ gcc gettext libasound2-dev \
+              libass-dev  libffms2-dev libfftw3-dev libfontconfig1-dev   \
+              libfribidi-dev libgtest-dev libhunspell-dev libicu-dev libopenal-dev       \
+              libpulse-dev libuchardet-dev libwxgtk3.0-gtk3-dev luarocks portaudio19-dev \
+              python3-pip python3-setuptools \
+              pkg-config libgl1-mesa-dev libgl-dev libfreetype6-dev libfontconfig-dev \
+              intltool libboost-all-dev
```

Comment:
CMake and autoconf builds have different config requirements.
- autoconf: configures and builds fine with only the following boost libs:
  `libboost-dev libboost-chrono-dev libboost-filesystem-dev libboost-locale-dev libboost-locale-dev libboost-regex-dev libboost-system-dev libboost-thread-dev`
- CMake: on the other hand requires more libs and otherwise fails during configure (so I switched to `libboost-all-dev`)

Given that the autoconf build also builds and tests fine with the more limited library set, I suspect the CMake script is misconfigured.

Reply:
List of boost libs required by CMake can be found at https://github.com/wangqr/Aegisub/blob/e35c16dc85bb2eb0c18d1c2ad0d42671d58773de/CMakeLists.txt#L509 .